src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+            return uniqueName;
+        }
+
+        private async Task<SyntaxNode> ExpandPropertyAndAddFieldAsync(
+            Document document, 
+            SyntaxNode property, 
+            SyntaxNode getAccessor, 
+            SyntaxNode setAccessor, 
+            SyntaxNode root, 
+            IPropertySymbol propertySymbol, 
+            string fieldName, 
+            SyntaxGenerator generator, 
+            CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+
+            // add statements to existing accessors
+            var getAccessorStatements = new SyntaxNode[] {
+                generator.ReturnStatement(
+                    generator.IdentifierName(fieldName)) };
+            var newGetAccessor = await AddStatementsToAccessorAsync(
+                document, 
+                getAccessor, 
+                getAccessorStatements, 
+                generator, 
+                cancellationToken).ConfigureAwait(false);
+
+            var setAccessorStatements = new SyntaxNode[] {
+                generator.ExpressionStatement(generator.AssignmentStatement(
+                    generator.IdentifierName(fieldName),
+                    generator.IdentifierName("value"))) };
+            var newSetAccessor = await AddStatementsToAccessorAsync(
+                document, 
+                setAccessor, 
+                setAccessorStatements, 
+                generator, 
+                cancellationToken).ConfigureAwait(false);
+
+            var newProperty = generator
+                .WithAccessorDeclarations(property, new SyntaxNode[] { newGetAccessor, newSetAccessor})
+                .WithAdditionalAnnotations(Formatter.Annotation)
+                .WithAdditionalAnnotations(new SyntaxAnnotation("property"));
+            newProperty = await Formatter.FormatAsync(newProperty, workspace).ConfigureAwait(false);
+            var newRoot = root.ReplaceNode(property, newProperty);
+
+            var newField = generator.FieldDeclaration(
+                fieldName, 
+                generator.TypeExpression(propertySymbol.Type), 
+                Accessibility.Private,
+                DeclarationModifiers.From(propertySymbol)).
+                    WithAdditionalAnnotations(Formatter.Annotation);
+            var newFieldNodes = SpecializedCollections.SingletonEnumerable(newField);
+            newProperty = newRoot.GetAnnotatedNodes("property").Single();
+            newRoot = newRoot.InsertNodesBefore(newProperty, newFieldNodes);
+            return newRoot;
+        }
+
+        private async Task<SyntaxNode> AddStatementsToAccessorAsync(
+            Document document,  
+            SyntaxNode accessor,
+            SyntaxNode[] statements,
+            SyntaxGenerator generator, 
+            CancellationToken cancellationToken)
+        {
+            // shell to lang specific to update the accessor.
```

Comment:
Not sure this comment provides value.  Also, we normally don't call it "shelling out" for overrides in abstract classes.